docs/2.8.0/docs/canton/usermanual/kms/external_key_storage_aws.rst

```diff
@@ -0,0 +1,199 @@
+..
+   Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates.
+..
+   Proprietary code. All rights reserved.
+
+.. _external_key_storage_aws:
+
+Configure External Key Storage and Usage with a AWS KMS
+=======================================================
+
+.. enterprise-only::
+
+The following section describes the steps needed to enable :ref:`External Key Storage and Usage <kms_external_architecture>`
+in Canton using AWS KMS. These steps include configuring AWS KMS, as well as, configuring this particular mode of operation.
+
+.. note::
+    Nodes configured with and without external KMS key storage can interact with each other,
+    including nodes using different KMS providers, as long as they follow the same crypto schemes.
+
+.. _external_key_storage_permissions_aws:
+
+AWS KMS Configuration
+---------------------
+
+To start using this feature we need to first enable a KMS for Canton.
+
+:ref:`--Configure AWS KMS for Canton-- <kms_aws_setup>`
+
+The following IAM permissions are required:
+
+- `kms:CreateKey`
+- `kms:TagResource`
+- `kms:Decrypt`
+- `kms:Sign`
+- `kms:DescribeKey`
+- `kms:GetPublicKey`
+
+When you are using cross-account keys, you do not need the `kms:CreateKey` and `kms:TagResource` permissions.
+
+External Key Storage and Usage Configuration
+--------------------------------------------
+
+External key storage and usage support can be enabled for a new installation (i.e., during the node
+bootstrap) or for an existing deployment.
+**Be aware that if a node has already been deployed you need to** :ref:`perform a node migration <participant_aws_kms_migration>`.
+Simply adding the following configuration is not enough.
+
+In the example below, we configure **a new** Canton participant node (called ``participant1``) to generate and
+store private keys in an external AWS KMS. The same configuration is applicable for all other node entities, e.g. domain-manager,
+mediators, sequencers.
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/resources/aws-kms-provider-tagged.conf
+   :language: none
+   :start-after: user-manual-entry-begin: KmsProviderConfig
+   :end-before: user-manual-entry-end: KmsProviderConfig
+
+An example configuration that puts together both AWS KMS and external storage configuration is shown below:
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/resources/aws-kms-provider-tagged.conf
+   :language: none
+   :start-after: user-manual-entry-begin: AwsKmsProviderFullConfig
+   :end-before: user-manual-entry-end: AwsKmsProviderFullConfig
+
+By default nodes run a ``tink`` provider that is not compatible with KMS provider. If you want to continue to have other nodes
+running a `non-KMS Canton` you are obliged to use a ``jce`` provider
+and you must explicitly configure it to use the KMS supported algorithms as the required algorithms. Here is an
+example for a domain:
```

Comment:
This paragraph assumes that the reader already has nodes running with KMS which is not necessarily the case. It's even at a bit at odds with the section which starts with
 
```
In the example below, we configure **a new** Canton participant node
```

This is of course important information but it's a bit confusing IMO that in the middle of the setup guide for AWS KMS on a participant I'm being told about tink being a default provider and that I need to configure jce on my domain. I'd put this in a separate sub-section, called something like "Interoperability with other nodes"